sdks/python/apache_beam/transforms/core.py

```diff
@@ -297,6 +298,51 @@ def get_function_arguments(obj, func):
   return getfullargspec(f)
 
 
+class RunnerAPIProtoHolder(object):
+  """An object for holding runner API protos for constructing a Python pipeline.
+
+  This is used for transforms, coders, etc. for which corresponding objects
+  cannot be initialized in Python SDK. For example, for `ParDo` transforms for
+  remote SDKs that may be available in Python SDK transform graph when expanding
+  a cross-language transform since a Python `ParDo` object cannot be generated
+  without a serialized Python `DoFn` object.
+  """
+
+  def __init__(self, proto):
+    self._proto = proto
+
+  def proto(self):
+    """Runner API payload for constructing the Python pipeline construct."""
+    return self._proto
+
+
+class RunnerAPIPTransformHolder(RunnerAPIProtoHolder, PTransform):
+  """A `RunnerAPIProtoHolder` for `PTransform`s"""
+
+  def to_runner_api(self, context, has_parts=False):
+    return self._proto
+
+  @staticmethod
+  def is_external_transform(proto):
+    if proto.urn != python_urns.PICKLED_TRANSFORM:
+      return True
+
+
+class RunnerAPICoderHolder(RunnerAPIProtoHolder, Coder):
+  """A `RunnerAPIProtoHolder` for `Coder`s"""
+
+  def to_runner_api(self, context):
+    return self._proto
+
+  def to_type_hint(self):
+    return typehints.Any
+
+  @staticmethod
+  def is_external_coder(proto):
+    if proto.spec.spec.urn != python_urns.PICKLED_CODER:
```

Comment:
This isn't correct, e.g. all the standard coders. Similarly for is_external_transform. I'd just remove those methods.